addons/website_forum/static/src/js/website_forum.js

```diff
@@ -292,43 +295,45 @@ publicWidget.registry.websiteForum = publicWidget.Widget.extend({
      * @private
      * @param {Event} ev
      */
-    _onFlagAlertClick: function (ev) {
-        var self = this;
+    _onFlagAlertClick: async function (ev) {
         ev.preventDefault();
-        var $link = $(ev.currentTarget);
-        this._rpc({
-            route: $link.data('href') || ($link.attr('href') !== '#' && $link.attr('href')) || $link.closest('form').attr('action'),
-        }).then(function (data) {
-            if (data.error) {
-                var message;
-                if (data.error === 'anonymous_user') {
-                    message = _t("Sorry you must be logged to flag a post");
-                } else if (data.error === 'post_already_flagged') {
-                    message = _t("This post is already flagged");
-                } else if (data.error === 'post_non_flaggable') {
-                    message = _t("This post can not be flagged");
-                }
-                this.displayNotification({
-                    message: message,
-                    title: _t("Access Denied"),
-                    sticky: false,
-                    type: "warning",
-                });
-            } else if (data.success) {
-                var elem = $link;
-                if (data.success === 'post_flagged_moderator') {
-                    elem.data('href') && elem.html(' Flagged');
-                    var c = parseInt($('#count_flagged_posts').html(), 10);
-                    c++;
-                    $('#count_flagged_posts').html(c);
-                } else if (data.success === 'post_flagged_non_moderator') {
-                    elem.data('href') && elem.html(' Flagged');
-                    var forumAnswer = elem.closest('.forum_answer');
-                    forumAnswer.fadeIn(1000);
-                    forumAnswer.slideUp(1000);
-                }
```

Comment:
All this diff for literally only 3 lines of real diff..
(you can see it in `?w=1` in the URL)
I don't like such big diff that just do nothing but why not.. I would have at least put it in a separate commit "Clean code before modify it" but whatever